qubes/ext/gui.py

```diff
@@ -121,22 +117,15 @@ def on_domain_start(self, vm, event, **kwargs):
             attached_vm.untrusted_qdb.write('/qubes-gui-domain-xid',
                                             str(vm.xid))
 
-    @qubes.ext.handler('domain-feature-pre-set:keyboard-layout')
-    def on_feature_pre_set(self, subject, event, feature, value, oldvalue=None):
-        untrusted_xkb_layout = value.split('+')
-        if len(untrusted_xkb_layout) != 3:
-            raise qubes.exc.QubesValueError("Invalid number of parameters")
-
-        untrusted_layout = untrusted_xkb_layout[0]
-        untrusted_variant = untrusted_xkb_layout[1]
-        untrusted_options = untrusted_xkb_layout[2]
-
-        re_variant = r'^[a-zA-Z0-9-_]*$'
-        re_options = r'^[a-zA-Z0-9-_:,]*$'
-
-        if not untrusted_layout.isalpha():
-            raise qubes.exc.QubesValueError("Invalid layout provided")
-        if not re.match(re_variant, untrusted_variant):
-            raise qubes.exc.QubesValueError("Invalid variant provided")
-        if not re.match(re_options, untrusted_options):
-            raise qubes.exc.QubesValueError("Invalid options provided")
+    @qubes.ext.handler('property-reset:keyboard_layout')
+    def on_keyboard_reset(self, subject, event, name, oldvalue=None):
+        if getattr(subject, 'guivm'):
+            kbd_layout = subject.guivm.keyboard_layout
+        else:
+            kbd_layout = 'us++'
```

Comment:
Since the logic for getting the default is already embedded into the property, you can simply use `subject.keyboard_layout` here.